### PR TITLE
198 - Extract instructor name feature has been updated

### PR DESCRIPTION
**Pull request by qinj15**
## Description

a new feature for extracting the Professor's name from a PDF syllabus file, and the PoC has been updated.

**Issue Link**: 
#198 
## Testing

Please mention the verification/testing used for this PR, which helps others to verify (**please remove this**).

## Type of action

- [ ] Bug fix 
- [x] New feature
- [ ] Requires a documentation update
- [ ] Others: 

## Checklist

- [ ] My code style is consistent with the current project
- [ ] I have done a brief review of my code
- [ ] I have written readable comments
- [ ] I have updated the documentation related to the changes I made
- [ ] I have written reasonable tests and the updated version all passed
- [ ] There should be no potential errors after the changes get merged into main

# Action Required

Please mention the further steps needed to do if necessary  (**please remove this**).

### CODE REVIEW

**Review by gaoq20 — APPROVED** [review]

File: src/app.py [hunk 1]
```diff
@@ -263,26 +264,33 @@ def course_detail(course_id):
...
+        instructor_name = extract_instructor_name_from_pdf(
+            pdf_name,
             bucket_name,
             s3,
         )
     else:
         email_list = []
+        instructor_name = ""
+
+    print("prof:", instructor_name)
```
Comment (katrina799) [inline src/app.py]: Please remove this print line

File: src/app.py [hunk 2]
```diff
@@ -311,7 +319,25 @@ def upload_file(course_id):
             file, bucket_name, file.filename, ExtraArgs={"ACL": "private"}
         )
 
-        update_csv(course_id, file.filename)
+        # 提取电子邮件列表
```
Comment (katrina799) [inline src/app.py]: Please change this comment

File: src/app.py [hunk 3]
```diff
@@ -311,7 +319,25 @@ def upload_file(course_id):
...
+                pdf_name,
+                bucket_name,
+                s3,
+            )
+        else:
+            email_list = []
+            instructor_name = ""
+
+        update_csv(course_id, file.filename, email_list, instructor_name)
         print("returning")
```
Comment (katrina799) [inline src/app.py]: Please remove these print lines

File: src/helper.py [hunk 4]
```diff
@@ -115,16 +115,28 @@ def upload_df_to_s3(df, s3, bucket_name, s3_csv_file_path):
     os.remove(new_csv_file_path)
 
 
-def update_csv(course_id, pdf_name):
+def update_csv(course_id, pdf_name, email_list, instructor_name):
     csv_file_path = "./poc-data/mock_course_info.csv"
```
Comment (katrina799) [inline src/helper.py]: Please make this a variable defined in `config.py`

File: src/helper.py [hunk 5]
```diff
@@ -145,23 +157,45 @@ def check_syllabus_exists(course_id, s3, bucket_name):
...
+
+    pdf_reader = pypdf.PdfReader(pdf_file_obj)
+    text = ""
+
+    for page_num in range(len(pdf_reader.pages)):
+        page = pdf_reader.pages[page_num]
+        text += page.extract_text()
+
+    # 查找 “Dr.” 或 “Instructor:” 后面的两个单词组成的名字
+    # 这里的正则表达式匹配 “Dr.” 或 “Instructor:” 后的任意空格，然后是两个单词
```
Comment (katrina799) [inline src/helper.py]: Please change comments here

File: docs/DevelopmentPlan/DevelopmentPlan.tex [hunk 6]
```diff
@@ -139,7 +139,7 @@ \subsection*{What will be the main risk for the success of the project?}
     In our project, we need to:
     \begin{itemize}
         \item Manage a database
-        \item Implement an algorithm that extracts date and task-related texts from a PDF
+        \item Implement an algorithm that course information texts from a PDF
```
Comment (katrina799) [inline docs/DevelopmentPlan/DevelopmentPlan.tex]: Implement an algorithm that `extracts` course information texts from a PDF?

**Review by katrina799 — APPROVED** [review]

**Review by songc120 — APPROVED** [review]

**Pull request merged**